smu.c

```diff
@@ -232,7 +232,7 @@ dolineprefix(const char *begin, const char *end, int newblock) {
 		}
 
 		/* Skip empty lines in block */
-		while(*(buffer + j - 1) == '\n') {
+		while(buffer + j - 1 >= buffer && *(buffer + j - 1) == '\n') {
```

Comment:
@ptrstr Pointer arithmetic is only valid as long as the result is within the array bound *or one past it*. [^0]

[^0]: https://port70.net/~nsz/c/c11/n1570.html#6.5.6p8

> If both the pointer operand and the result point to elements of the
> same array object, or one past the last element of the array object,
> the evaluation shall not produce an overflow; otherwise, the behavior
> is undefined.

So doing `buffer - 1` is not correct. The proper fix would be to just check if `j-1 > 0 && j-1 <= sizeof buffer` before doing the arithmetic: `buffer + (j-1)`.